### PR TITLE
Improve UI debugging

### LINKED ISSUE #218: Create debug UI for shared object manager
We have the debug menu already, we should try to add in a window for our shared object manager with filters/duplication finders so we can easily debug what all is being tracked. 

We could then add features like clicking on a party in that debug list would center the camera on it. Would be helpful for verification purposes that all parties are truly synced between clients/server.

### PR DESCRIPTION
**Pull request by thomasDelaporte**
## PR Checklist

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is new 
- Open/close (toggle) of the debugging UI through keys
- Visibility of all objects saved in the CoopObjectManager class

Resolves #218 
New Debug UI

### CODE REVIEW

**Comment by garrettluskey** [comment]
Freezes on client for some reason. Please fix

File: source/Coop/Mod/DebugUtil/DebugUI.cs [hunk 1]
```diff
@@ -49,11 +53,99 @@ public void Update(TimeSpan frameTime)
                 DisplayConnectionInfo();
                 DisplayHarmonyPatches();
                 DisplayPersistenceMenu();
+                DisplayCoopObjectManagers();
                 End();
+
+                foreach(object detailObject in this.detailsObjects)
```
Comment (garrettluskey) [inline source/Coop/Mod/DebugUtil/DebugUI.cs]: Move block to independant section

File: source/Coop/Mod/DebugUtil/DebugUI.cs [hunk 2]
```diff
@@ -49,11 +53,99 @@ public void Update(TimeSpan frameTime)
...
+                    this.detailsObjects.Remove(toDeleteObject);
+                }
+
+                this.toDeleteObjects.Clear();
+                Imgui.EndMainThreadScope();
             }
         }
         public int Priority { get; } = UpdatePriority.MainLoop.DebugUI;
 
+        private void DisplayCoopObjectManagers()
```
Comment (garrettluskey) [inline source/Coop/Mod/DebugUtil/DebugUI.cs:85]: Add comments

**Review by garrettluskey — CHANGES_REQUESTED** [review]
Fix on client and update code, then will be good.

**Comment by thomasDelaporte** [comment]
@garrettluskey As we quickly discussed, the freeze is probably due to the gamedata and not the code quoted here.

I cleaned up the class as much as possible and added a lot of documentation for the whole class. 

The problem now is that the client has no data in CoopObjectManager so no way to check if it matches on both sides. The persistence => parties tab is also empty in the client.

**Review by garrettluskey — APPROVED** [review]
Nice

**Pull request merged**